lark/parser_frontends.py

```diff
@@ -23,12 +23,18 @@ def get_frontend(parser, lexer):
         elif lexer == 'contextual':
             return LALR_ContextualLexer
         elif issubclass(lexer, Lexer):
+            class CustomLexerWrapper(Lexer):
```

Comment:
This would prevent people from actually implementing the new interface (like I am doing over at lark-language-server). I would suggest expecting till 1.0 a `new_sytle` class-attribute that allows the lexer to use the new interface if set to true. After 1.0, old style will be unsupported.

I would also suggest adding a deprecation warning.

Reply:
Yes, I plan to change this in v1.0

How would you prefer to expose the new interface until then?

Reply:
Have a class attribute (maybe `new_style`?) that, when present and true makes this not wrap the class.